src/economy/shops/taxes.ts

```diff
@@ -0,0 +1,228 @@
+  'tax',
+  'taxes',
+  'taxchest',
+  'tax chest',
+  'vero',
+  'veroarkku',
+  'verot',
+]);
+
+/**
+ * Check if the sign and the block (chest) behind it can become a tax chest
+ * @param sign Sign in front of the tax chest
+ * @param lines Lines of the sign (From SignChangeEvent)
+ * @returns Boolean (Can the chest become a tax chest or not)
+ */
+function canBecomeTaxChest(sign: Block, lines: string[]) {
+  if (!sign) return false;
+  if (!(sign.state instanceof Sign)) return false;
+
+  // Check if lines are either "tax", "veroarkku", "tax chest" etc or empty
+  for (const line of lines) {
+    if (line && !taxWords.has(line.toLowerCase())) return false;
+  }
+  if (lines.every((line) => !line)) return false;
+
+  if (!(sign.blockData instanceof WallSign)) return false;
+  const attachedTo = getBlockBehind(sign);
+  if (!attachedTo) return false;
+  if (!(attachedTo.state instanceof Chest)) return false;
+  return true;
+}
+
+/**
+ * Add untransferred taxes to the tax chest
+ */
+registerEvent(PlayerInteractEvent, (event) => {
+  if (event.action !== Action.RIGHT_CLICK_BLOCK) return;
+  if (!(event.clickedBlock?.state instanceof Chest)) return;
+  const chest = event.clickedBlock.state;
+  const taxChestView = dataView(TaxChestData, chest);
+  if (!taxChestView.taxCollector) return;
+  const player = event.player;
+
+  const collector = Bukkit.getOfflinePlayer(
+    UUID.fromString(taxChestView.taxCollector),
+  );
+  const collectorView = dataView(TaxCollectorData, collector);
+
+  const yupLoc = collectorView.chestLocation;
+  const location = yupLocToLoc(yupLoc);
+  if (!location) return;
+
+  if (distanceBetween(location, chest.location)) {
+    errorMessage(player, 'Tämä veroarkku ei ole enää käytössä!');
+    return;
+  }
+
+  player.sendMessage(
+    color('#FFAA00', text('-----------------Veroarkku-----------------')),
+  );
+  player.sendMessage(
+    color('#FFFF99', text('Veronkerääjä: ' + ChatColor.GOLD + collector.name)),
+  );
+
+  if (collectorView.untransferred.length) {
+    player.sendMessage(color('#FFFF99', text('Uusia veroja:')));
+    for (const tax of collectorView.untransferred) {
+      const currency = getShopCurrency(tax.currency);
+      if (!currency) return;
+      giveMoney(chest.inventory, tax.amount, currency);
+      player.sendMessage(
+        color(
+          '#FFFF99',
+          text(
+            '+ ' +
+              ChatColor.GOLD +
+              round(tax.amount, 2) +
+              ' ' +
+              ChatColor.RESET +
+              currency.unitPlural,
+          ),
+        ),
+      );
+    }
+
+    collectorView.untransferred = [];
+  }
+
+  player.sendMessage(
+    color('#FFAA00', text('-------------------------------------------')),
+  );
+});
+
+/**
+ * Check if the player is a tax collector or not
+ * @param player Player to be checked
+ * @returns The tax collector as OfflinePlayer or undefined
+ */
+export function getTaxCollector(player: string | OfflinePlayer) {
```

Comment:
It is usually better to avoid this kind of "dual-use" functions. isDataCollection(OfflinePlayer): boolean would be cleaner; the caller can resolve offline player from string quite easily.